IThenticate.inc.php

```diff
@@ -0,0 +1,438 @@
+        
+        // Validate the EULA version
+        if (!$this->validateEulaVersion($this->eulaVersion ?? $version)) {
+            return false;
+        }
+
+        // if user has already accepted/confirm this EULA, no need to go further to reconfirm
+        if ($this->verifyUserEulaAcceptance($user, $this->getApplicationEulaVersion())) {
+            return true;
+        }
+        
+        $response = Application::get()->getHttpClient()->request(
+            'POST',
+            $this->getApiPath("eula/{$this->getApplicationEulaVersion()}/accept"),
+            [
+                'headers' => array_merge($this->getRequiredHeaders(), [
+                    'Content-Type' => 'application/json'
+                ]),
+                'json' => [
+                    'user_id' => $user->getId(),
+                    'accepted_timestamp' => \Carbon\Carbon::now()->toIso8601String(),
+                    'language' => $this->getEulaConfirmationLocale($context->getPrimaryLocale()),
+                ],
+                'verify' => false,
+                'exceptions' => false,
+            ]
+        );
+
+        return $response->getStatusCode() === 200;
+    }
+    
+    /**
+     * Create a new submission at service's end
+     * @see https://developers.turnitin.com/docs/tca#create-a-submission
+     * 
+     * @param Submission    $submission
+     * @param User          $user
+     * @param Author|null   $author
+     * @param Site|null     $site
+     *
+     * @return string|null  if succeed, it will return the created submission UUID at service's end and 
+     *                      at failure, will return null
+     */
+    public function submitSubmission($submission, $user, $author = null, $site = null) {
+
+        $publication = $submission->getCurrentPublication(); /** @var Publication $publication */
+        $author ??= $publication->getPrimaryAuthor();
+
+        if (!$site) {
+            import('lib.pkp.classes.db.DAORegistry');
+            $siteDao = DAORegistry::getDAO("SiteDAO"); /** @var SiteDAO $siteDao */
+            $site = $siteDao->getSite(); /** @var Site $site */
+        }
+
+        $response = Application::get()->getHttpClient()->request(
+            'POST',
+            $this->getApiPath("submissions"),
+            [
+                'headers' => array_merge($this->getRequiredHeaders(), [
+                    'Content-Type' => 'application/json'
+                ]),
+                'json' => [
+                    'owner' => $author->getId(),
+                    'title' => $publication->getLocalizedTitle($publication->getData('locale')),
+                    'submitter' => $user->getId(),
+                    'metadata' => [
+                        'owners' => [
+                            [
+                                'id' => $author->getId(),
+                                'given_name' => $author->getGivenName($publication->getData('locale')) ?? $author->getLocalizedGivenName(),
```

Comment:
The `??` clause here and on the next line are not necessary; we can guarantee that the given name will always be present in the publication locale (as it's required), and we should not risk mixing parts of names from different locales.